datamol/io.py

```diff
@@ -95,20 +97,28 @@ def read_sdf(
             `as_df` is True.
         include_computed: Include computed properties in the columns.  Only relevant if
             `as_df` is True.
+        sanitize: Whether to sanitize the molecules
+        strictParsing: If set to false, the parser is more lax about correctness of the contents.
     """
 
     # File-like object
     if isinstance(urlpath, io.IOBase):
-        supplier = Chem.ForwardSDMolSupplier(urlpath)
-        mols = [mol for mol in supplier if mol is not None]
+        supplier = Chem.ForwardSDMolSupplier(urlpath, sanitize=False, strictParsing=strictParsing)
+        if sanitize == True:
+            mols = [dm.sanitize_mol(mol) for mol in supplier if mol is not None]
+        else:
+            mols = [mol for mol in supplier if mol is not None]
 
     # Regular local or remote paths
     else:
         with fsspec.open(urlpath) as f:
             if str(urlpath).endswith(".gz") or str(urlpath).endswith(".gzip"):
                 f = gzip.open(f)
-            supplier = Chem.ForwardSDMolSupplier(f)
-            mols = [mol for mol in supplier if mol is not None]
+            supplier = Chem.ForwardSDMolSupplier(f, sanitize=False, strictParsing=strictParsing)
+            if sanitize == True:
+                mols = [dm.sanitize_mol(mol) for mol in supplier if mol is not None]
+            else:
+                mols = [mol for mol in supplier if mol is not None]
 
```

Comment:
In the `if-else`, you are repeating the block `if sanitize == True`. I would put it outside just before the `if as_df:`.